### PR TITLE
Check vendor sync

### PR DESCRIPTION
**Pull request by ganeshmaharaj**
dep ensure -dry-run -no-vendor will check and show if vendor code is out
of sync with GOpkg files but always has a return code of 0. Switching it
to dep check will fail if Gopkg is not in sync with vendor.

Fixes: #1794
Signed-off-by: Ganesh Maharaj Mahalingam <ganesh.mahalingam@intel.com>

### CODE REVIEW

**Comment by grahamwhaley** [comment]
can you split the `dep` static check change and the actual re-vendor into two separate commits pls ;-)

**Comment by jodh-intel** [comment]
This PR also fixes #1733 fwics? But I'm unclear about the vendor changes here:

- this PR changes a bunch of vendored code but no change to `Gopkg.toml`?
- If these changes are actually needed (?), please could you document what changed in [the usual way](https://github.com/kata-containers/community/blob/master/CONTRIBUTING.md#re-vendor-prs).

**Comment by ganeshmaharaj** [comment]
@jodh-intel the vendor change is just by running `dep ensure` in the repository. It seems we have had Gopkg.toml change in the past but the match vendor update has not happened along with the code. I can try and find all the contents that changed, but it seems this run is an old tech debt.

**Comment by jodh-intel** [comment]
We only re-vendor *"when we need to"* meaning:

- when we want new features.
- when we notice important upstream fixes.

And if we re-vendor we need to know what changed in case this "breaks the world".

**Comment by ganeshmaharaj** [comment]
THis PR has brought up a topic that we might need to discuss in the next AC meeting. A decision will be made after this has been discussed.

**Comment by ganeshmaharaj** [comment]
@jodh-intel @grahamwhaley Sorry i missed the next AC meeting. Do you happen to have an update on this from that meeting?

**Comment by jodh-intel** [comment]
@ganeshmaharaj - we deferred it until the next meeting ;)

**Comment by ganeshmaharaj** [comment]
@jodh-intel @amshinde As discussed, while we wait for go mod to become mainline, this is a quick cleanup of our dependency and make `dep` more reliable. We are dumping all `Gopkg.*` files other than the one in the top folder.

**Comment by ganeshmaharaj** [comment]
@chavafg @GabyCT @jcvenegas PTAL?

File: Gopkg.toml [hunk 1]
```diff
@@ -44,3 +44,45 @@
...
+  version = "1.3.0"
+
+[[constraint]]
+  name = "gopkg.in/russross/blackfriday.v2"
+  version = "2.0.1"
+
+[[constraint]]
+  name = "github.com/slok/kubewebhook"
+  version = "0.2.0"
+
```
Comment (amshinde) [inline Gopkg.toml]: nit: extra spaces

File: Gopkg.toml [hunk 2]
```diff
@@ -44,3 +44,45 @@
...
+
+[[constraint]]
+  branch = "master"
+  name = "k8s.io/api"
```
Comment (amshinde) [inline Gopkg.toml]: I was going to say we should pin these to a specific version as these are currently pulled from the master branch, but looks like these were never pinned. 
I am not sure if the tool itself required the latest, something to consider.

**Comment by chavafg** [comment]
looks good, but travis failed on `macos` job:

```
go build -o "kata-check-markdown" -ldflags "-X main.name=kata-check-markdown -X main.commit="9012bca07e8bc0b9a6897d479efbc639543e2f64-dirty" -X main.version=0.0.1" .
# github.com/kata-containers/tests/vendor/github.com/sirupsen/logrus
../../vendor/github.com/sirupsen/logrus/logger.go:20:8: undefined: LevelHooks
make: *** [kata-check-markdown] Error 2
The command "bash .ci/static-checks.sh github.com/kata-containers/tests" exited with 2.
```

**Comment by GabyCT** [comment]
Travis is failing with
```
../../vendor/github.com/sirupsen/logrus/logger.go:20:8: undefined: LevelHooks
make: *** [kata-check-markdown] Error 2
The command "bash .ci/static-checks.sh github.com/kata-containers/tests" exited with 2.
```

File: Gopkg.toml (hunk 2 above)
Comment (ganeshmaharaj) [reply Gopkg.toml]: Agreed. Not sure if this was intentional to make sure we always keep up with latest API changes from k8s. @mcastelino ?

**Comment by ganeshmaharaj** [comment]
@raravena80 any clue on why travis is failing for OSX? i am not able to make sense of the failure and don't have a OSX machine with me to be able to test this.

File: Gopkg.toml (hunk 1 above)
Comment (ganeshmaharaj) [reply Gopkg.toml]: ohh yeah. let me fix it.

**Comment by jodh-intel** [comment]
It's something to do with the (old) sub-vendoring but, like you, I don't have access to a MacBook (clearly we're not hip enough :cry:)

**Pull request merged**